File: LeetCode/0101.Symmetric-Tree/Symmetric-Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
bool isTwoTreeSymmetric(struct TreeNode* a, struct TreeNode* b) {
    if (a==NULL && b==NULL) return true;
    if (a==NULL || b==NULL) return false;
    if (a->val == b->val) {
        return isTwoTreeSymmetric(a->left, b->right) && isTwoTreeSymmetric(a->right, b->left);
    }
    return false;
}
bool isSymmetric(struct TreeNode* root) {
    if (root) return  isTwoTreeSymmetric(root->left, root->right);
    return true;
}

File: LeetCode/0101.Symmetric-Tree/Symmetric-Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        def isMirror(node1: TreeNode, node2: TreeNode):
            if node1 == node2 is None:
                return True
            if node1 is None or node2 is None:
                return False

            return all(((node1.val == node2.val),
                       isMirror(node1.left, node2.right),
                       isMirror(node1.right, node2.left)))
        return isMirror(root, root)
